rxbinding-recyclerview-v7/src/main/java/com/jakewharton/rxbinding/support/v7/widget/RecyclerAdapterDataChangeOnSubscribe.java

```diff
@@ -0,0 +1,45 @@
+package com.jakewharton.rxbinding.support.v7.widget;
+
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.RecyclerView.AdapterDataObserver;
+
+import com.jakewharton.rxbinding.internal.MainThreadSubscription;
+
+import rx.Observable;
+import rx.Subscriber;
+
+import static com.jakewharton.rxbinding.internal.Preconditions.checkUiThread;
+
+final class RecyclerAdapterDataChangeOnSubscribe<T extends RecyclerView.Adapter<? extends RecyclerView.ViewHolder>>
+        implements Observable.OnSubscribe<T> {
+    private final T adapter;
+
+    public RecyclerAdapterDataChangeOnSubscribe(T adapter) {
+        this.adapter = adapter;
+    }
+
+    @Override
+    public void call(final Subscriber<? super T> subscriber) {
+        checkUiThread();
+        final AdapterDataObserver observer = new RecyclerView.AdapterDataObserver() {
+            @Override
+            public void onChanged() {
```

Comment:
Only for item changes i think ,  incremental changes are these for when items are added , inserted .. (onItemRangeInserted, onItemRangeRemoved ..) ?